src/Api/FingerprintApi.php

```diff
@@ -126,57 +128,192 @@ public function getEvent($request_id)
             try {
                 $serialized = ObjectSerializer::deserialize($responseBody, $returnType, []);
             } catch (\Exception $e) {
-                throw new SerializationException($response);
+                throw new SerializationException($response, $e);
             }
 
             return [$serialized, $response];
+        } catch (ApiException $e) {
+            try {
+                switch ($e->getCode()) {
+                    case 200:
+                        /** @var ResponseInterface $response */
+                        $response = $e->getResponseObject();
+                        $responseBody = $response->getBody()->getContents();
+                        $response->getBody()->rewind();
+                        $data = ObjectSerializer::deserialize(
+                            $responseBody,
+                            '\Fingerprint\ServerAPI\Model\EventResponse',
+                            $response->getHeaders()
+                        );
+                        $e->setResponseObject($data);
+
+                        break;
+
+                    case 403:
+                        /** @var ResponseInterface $response */
+                        $response = $e->getResponseObject();
+                        $responseBody = $response->getBody()->getContents();
+                        $response->getBody()->rewind();
+                        $data = ObjectSerializer::deserialize(
+                            $responseBody,
+                            '\Fingerprint\ServerAPI\Model\ErrorEvent403Response',
+                            $response->getHeaders()
+                        );
+                        $e->setResponseObject($data);
+
+                        break;
+
+                    case 404:
+                        /** @var ResponseInterface $response */
+                        $response = $e->getResponseObject();
+                        $responseBody = $response->getBody()->getContents();
+                        $response->getBody()->rewind();
+                        $data = ObjectSerializer::deserialize(
+                            $responseBody,
+                            '\Fingerprint\ServerAPI\Model\ErrorEvent404Response',
+                            $response->getHeaders()
+                        );
+                        $e->setResponseObject($data);
+
+                        break;
+                }
+
+                throw $e;
+            } catch (\Exception $_) {
```

Comment:
Why are we not catching and throwing `SerializationException` here?

Reply:
We can't throw SerializationException from ObjectSerializer::deserialize because SerializationException expects ResponseInterface; we have this response only in API, so we throw an exception in ObjectSerializer and catch it in API, and we throw SerializationException with a response. ObjectSerializer only throws one type of exception, so there is no complexity for exceptions in ObjectSerializer. If you have another approach, please let me know again. I'll mark this resolved, but you can "unresolve" it anytime.
@TheUnderScorer